domain/src/main/java/org/oppia/android/domain/devoptions/ShowAllHintsAndSolutionHelper.kt

```diff
@@ -0,0 +1,19 @@
+package org.oppia.android.domain.devoptions
+
+import javax.inject.Inject
+import javax.inject.Singleton
+
+/** Monitor to identify whether all hints and solution should be shown. */
+@Singleton
+class ShowAllHintsAndSolutionHelper @Inject constructor() {
+
+  private var showAllHintsAndSolution = false
+
+  /** Returns [showAllHintsAndSolution] indicating whether showing all hints and solution feature is enabled. */
```

Comment:
This variable isn't accessible to outside users, so this documentation comment may actually be confusing. Suggest isntead:

```suggestion
  /** Returns whether showing all hints and solutions should be shown by default. */
```

Also, the way this was worded before was slightly confusing: this isn't gating hints & solutions, it's gating whether hints & solutions should be always shown. Please update variable names/method names/documentation in this PR accordingly to be clearer on this distinction.

Reply:
Updated Kdocs. PTAL. Not sure what variables/methods are you referring to here.